analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -93,6 +93,17 @@ public void Execute()
         checks.ExecutionCompleted();
     }
 
+    private bool CheckVisitCount(ExplodedNode current, int visitCount)
+    {
+        return visitCount <= MaxOperationVisits
+            || (visitCount <= MaxOperationVisits + 1 && IsLoopCondition(current));
+
+        bool IsLoopCondition(ExplodedNode current) =>
+            current.Block.ConditionalSuccessor is not null  // avoid further checks for non-conditional branching blocks
+            && (current.Operation?.Instance ?? current.Block.BranchValue) is { } branchValueOperation
```

Comment:
I find this very cryptic. It says, "give me any operation, or branch value on branching", but what it really wants to do is "if operation is part of branching, or it's branching, check that"
```
        bool IsLoopCondition() =>
            node.Block.BranchValue is not null
            && (node.Operation is null || IsInBranchValue(node.Operation.Instance))
            && syntaxClassifier.IsInLoopCondition(node.Block.BranchValue.Syntax);  // currently processing a loop condition

        bool IsInBranchValue(IOperation current)
        {
            while (current is not null)
            {
                if (current == node.Block.BranchValue)
                {
                    return true;
                }
                current = new IOperationWrapperSonar(current).Parent;
            }
            return false;
        }
```

Reply:
..it took me quite some debugging and experimenting time to decode the line